### PR TITLE
Fix JavaScript error messages using `reason` field instead of `message`

### PR DESCRIPTION
**Pull request by pvditto**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 /target
 **/*.rs.bk
 .DS_Store
+*.code-workspace
```
Comment (pvditto) [inline .gitignore:4]: I had to configure a multi-root code space to get rust-analyzer to work with all of the crates but I can also remove this .gitignore change if you think it's noisy, let me know.

**Comment by pvditto** [comment]
I ran into a wall when the removed `new` and `from_status` methods broke `derive_ReprC`. Can you give me a hint how to continue? @danielhenrymantilla 

```rs
error[E0599]: no function or associated item named `new` found for struct `safer_ffi::js::Error` in the current scope
   --> src/lib.rs:252:1
    |
252 | #[derive_ReprC(js)]
    | ^^^^^^^^^^^^^^^^^^^ function or associated item not found in `safer_ffi::js::Error`
```

**Comment by danielhenrymantilla** [comment]
Yeah, removing `Error::new()` is ending up to be more disruptive than anticipated; if our code does not rely on the `Status` in and of itself provided in the `new()` constructor then we may just ignore it on wasm, and let the safer-ffi-generated code still provide a `Status` to node.js code, I think

File: .gitignore (hunk 1 above)
Comment (danielhenrymantilla) [reply .gitignore:4]: Yeah don't worry this is fine 🙂

**Review by danielhenrymantilla — APPROVED** [review]
Thanks! ✅

**Pull request merged**